vllm/worker/worker.py

```diff
@@ -85,8 +86,10 @@ def __init__(
         ModelRunnerClass: Type[GPUModelRunnerBase] = ModelRunner
         if model_runner_cls is not None:
             ModelRunnerClass = model_runner_cls
-        elif self.model_config.embedding_mode:
+        elif self.model_config.model_mode == ModelMode.EMBEDDING:
```

Comment:
Perhaps we should let each mode override the `ModelRunnerClass` instead of handling each case inside `Worker`.

Reply:
I made some changes as you can see
But I didn't quite understand your point about "override BlockSpaceManagerImpl ModelRunnerClass". Do you mean to override the entire BlockManager/ModelRunner class? That seems more complicated than the current implementation.

Reply:
Here, by "override" I just mean set the `ModelRunnerClass` to use. We can add static methods to `ModelMode` (e.g. `get_block_space_manager_impl` and `get_model_runner_cls`) so we can keep the switch/case code all in one place.

Reply:
Please take a look class ModelMode, some 'circular import' problems occur, but solved.